go/vt/vtgate/engine/delete.go

```diff
@@ -250,6 +190,42 @@ func (del *Delete) deleteVindexEntries(vcursor VCursor, bindVars map[string]*que
 	return nil
 }
 
+func (del *Delete) deleteVindexEntriesScatter(vcursor VCursor, bindVars map[string]*querypb.BindVariable, rss []*srvtopo.ResolvedShard) error {
```

Comment:
Looks like this is a superset of `deleteVindexEntries`, and equally efficient. Let's delete that function and reuse this for all. This also means that this function cal be renamed to `deleteVindexEntries`.

Reply:
The current `deleteVindexEntries` takes `ksid` as an input argument, and `deleteVindexEntriesScatter` figures that out on its own. Is it OK to get the `ksid` from the result for the single shard `deleteVindexEntries` as well?